### DISCUSSION TITLE
pathdinfR output for RNAseq and DIA_proteomics data

### DISCUSSION THREAD

**Discussion by masaver** [egeulgen/pathfindR#63] [Q&A] [answered]
Hi Everyone,
I am studying pathways activity in one specific type of cancer, for which I have DIA_proteomics and RNAseq data, from normal and Tumor tissues.

For the proteomics data, I obtained the pvalues for differential expression using limma. For the RNAseq data, I used DESeq2. In both data types, there are several thousands of proteins/genes that are differentially regulated using and FDR of < 0.05.

When I run pathfindR ( with default parameters ) on the proteomics data, and then get the pathway scores, the differences between normal and Tumors look like this:
![crc reactome scores](https://user-images.githubusercontent.com/25987663/101766988-e5a95880-3ae3-11eb-869a-aa952fd8332b.png)
(Rows = Pathways ; Columns = Samples)

However, when I run pathfindR ( with default parameters, or even with sig_gene_thr = 0.01 
and sig_gene_thr = 0.7 ), i get the "No Active Subnetworks" message.

So my questions for you guys are:
1) Is there any logical explanation for why this might happen ?
2) Are there any use cases of pathfindR with RNAseq data ?
3) How much can you lower sig_gene_thr & score_quan_thr , and still be "confident" in your results ? And how would you test for the "confidence" of your results ?

Many thanks in advance.

Best,

-Mathias

**Comment by egeulgen** [comment]
Hello Mathias,

I'm not sure about the exact message you're getting, but if there are no active subnetworks identified, the issue is usually a small number of input genes.

1. To better answer this question: what exactly is the message you get? How many differentially-expressed genes are there for your RNAseq data? Also, what version of pathfindR are you using? If you wouldn't mind sharing the data and the script you used for the RNAseq data, I could pinpoint any potential issues. 
2. We have used pathfindR in many cases using output from RNAseq data. In fact, any gene-associated p-value data can be analyzed with pathfindR.
3. The default options for filtering active subnetworks were determined based on analyses of multiple datasets. We're still working on novel ways to ensure pathfindR is providing high-confidence results. There is no straightforward answer to your question. Ideally, any subnetwork that contains at least 2 input genes might be biologically relevant from the active-subnetwork-oriented enrichment analysis perspective. Hence, you may even keep all such active subnetworks.
For testing the confidence of your results, you can (a) search for literature support or (b) perform experimental validation of the enriched pathways. 

Best,
-E

**Comment by masaver** [comment]
The pathfindR version I\m using is pathfindR_1.6.0

Also, from the differential expression analysis I saw that:
Down-Regulated genes = 10443
Up-Regulated gene =  14386
 Genes with no significant change = 8761

At the start of the analysis i get this message:
"## Testing input
The input looks OK
## Processing input. Converting gene symbols,
          if necessary (and if human gene symbols provided)
Number of genes provided in input: 33588
Number of genes in input after p-value filtering: 24827
pathfindR cannot handle p values < 1e-13. These were changed to 1e-13"

But then ,tat the end of the pathfindR analysis i get this message ( and
the results object/data.frame is empty ):
"Found 0 active subnetworks


Warning message:
Did not find any enriched terms!"

I run pathfindR with the following command:
rna.res.pathfindR = run_pathfindR( input = rna.pathfindR ,
                                     output_dir =
"../Results/rna.CRC.pathfindR_results.reactome" ,
                                     gene_sets = "Reactome" ,
                                     plot_enrichment_chart = FALSE ,
                                     visualize_enriched_terms = FALSE ,
                                     score_quan_thr = 0.7 ,
                                     sig_gene_thr = 0.01 )

Where  rna.pathfindR is the pathfindR input data.frame. I'm sending you a
copy of it, saved as a .rds file,

Let me know if you need more information .

Best,

-Mathias

**Comment by ozanozisik** [comment, marked as answer]
Hello Mathias,
In pathfindR, in line with the scoring in jActiveModules by Ideker et al., a background score distribution is calculated and it is used to adjust the score of subnetworks. In your case, almost all genes are significant, which prevents any subnetwork from being significant. I suggest using a more strict filtering on your gene set, taking logfoldchange into account.
Best,
Ozan

**Comment by egeulgen** [comment]
Hey @masaver,

As @ozanozisik pointed out, you should be filtering for significantly differentially expressed genes, taking into logFC account. Below are two volcano plots, showing p<0.05 only (left) and |LFC| > 1.5 + p < 0.05 (right). As you see, many of the genes in your RNAseq data have low logFC values, implying little impact (even if statistically significant).

![volcano_plots](https://user-images.githubusercontent.com/12850155/101800448-579e9380-3b1e-11eb-8ef7-bc236dd0e952.png)

After using the latter filtering approach (i.e., |LFC| > 1.5 + p < 0.05), I obtained 93 enriched pathways for your data (with default subnetwork filtering).

Hope these answers help,
Best,
-E

**Comment by masaver** [comment]
That was of great help!. Indeed, filtering the genes by |LFC| > 1.5 + p < 0.05 did the trick.
Thanks a lot for the Help.

Best,
-Mathias
